### PR TITLE
EF instrumentation modified to depend only on API, other refactorings.

### PR DESCRIPTION
**Pull request by cijothomas**

### CODE REVIEW

File: src/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore.csproj [hunk 1]
```diff
@@ -8,7 +8,7 @@
   </PropertyGroup>
```
Comment (utpilla) [inline src/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore.csproj:8]: We should add `<IncludeSharedInstrumentationSource>true</IncludeSharedInstrumentationSource>`. This project was depending on the SDK before to create `TracerProviderSdk`.

File: src/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore.csproj [hunk 2]
```diff
@@ -8,7 +8,7 @@
...
 
   <ItemGroup>
-    <PackageReference Include="OpenTelemetry" Version="1.0.1" />
+    <PackageReference Include="OpenTelemetry.Api" Version="1.0.1" />
```
Comment (utpilla) [inline src/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore.csproj:11]: `EntityFrameworkCore.Tests` project depends on Sdk for some of the unit tests. Do we want to add the SDK dependency in `OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore.Tests.csproj`?

**Review by utpilla — APPROVED** [review]

File: build/Common.props [hunk 3]
```diff
@@ -64,10 +64,10 @@
   </ItemGroup>
 
   <ItemGroup Condition="'$(IncludeSharedTestSource)'=='true'">
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Contrib.Shared\Test\TestExporter.cs" Link="Includes\TestExporter.cs" />
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Contrib.Shared\Test\TestSampler.cs" Link="Includes\TestSampler.cs" />
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Contrib.Shared\Test\TestActivityExportProcessor.cs" Link="Includes\TestActivityExportProcessor.cs" />
-    <Compile Include="$(RepoRoot)\src\OpenTelemetry.Contrib.Shared\Test\TestActivityProcessor.cs" Link="Includes\TestActivityProcessor.cs" />
+    <Compile Include="$(RepoRoot)\test\OpenTelemetry.Contrib.Tests.Shared\TestExporter.cs" Link="Includes\TestExporter.cs" />
```
Comment (srprash) [inline build/Common.props:67]: Nit: Should we call it "OpenTelemetry.COntrib.Shared.Tests" to be consistent with other projects?

File: src/OpenTelemetry.Contrib.Shared/OpenTelemetry.Contrib.Shared.csproj [hunk 4]
```diff
@@ -5,7 +5,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="OpenTelemetry" Version="1.1.0-beta1" />
+    <PackageReference Include="OpenTelemetry.Api" Version="1.1.0-beta1" />
```
Comment (srprash) [inline src/OpenTelemetry.Contrib.Shared/OpenTelemetry.Contrib.Shared.csproj:8]: Can we use v1.0.1 of OTel API here?

File: test/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCoreTests/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore.Tests.csproj [hunk 5]
```diff
@@ -18,6 +18,7 @@
 
   <ItemGroup>
     <ProjectReference Include="$(RepoRoot)\src\OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore\OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore.csproj" />
+    <PackageReference Include="OpenTelemetry" Version="1.1.0-beta1" />
```
Comment (srprash) [inline test/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCoreTests/OpenTelemetry.Contrib.Instrumentation.EntityFrameworkCore.Tests.csproj:21]: Use 1.0.1 here as well?

File: test/OpenTelemetry.Contrib.Tests.Shared/OpenTelemetry.Contrib.Tests.Shared.csproj [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <IsTestProject>false</IsTestProject>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="OpenTelemetry" Version="1.1.0-beta1" />
```
Comment (srprash) [inline test/OpenTelemetry.Contrib.Tests.Shared/OpenTelemetry.Contrib.Tests.Shared.csproj:9]: Here too?

File: build/Common.props (hunk 3 above)
Comment (cijothomas) [reply build/Common.props:67]: this was named Tests.Shared intentionally to not match the *Tests.dll* in the codecoverage gh action!

**Pull request closed without merging**